sympy/controlsys/lti.py

```diff
@@ -0,0 +1,1363 @@
+
+>>> ssm = StateSpaceModel(tfm)
+>>> tfm2 = TransferFunctionModel(ssm, s)
+TransferFunctionModel(Matrix([
+[         1/s],
+[s/(s**2 + 1)]]))
+```
+
+Or connect two system to create a new one:
+
+>>> var('a:3, b:3')
+(a0, a1, a2, b0, b1, b2)
+>>> ssm1 = StateSpaceModel(
+...         Matrix([[0, 1],
+...                [-a0, -a1]]),
+...         Matrix([0, 1]),
+...         Matrix([[b0, b1]]))
+>>> ssm1
+StateSpaceModel(
+Matrix([
+[  0,   1],
+[-a0, -a1]]),
+Matrix([
+[0],
+[1]]),
+Matrix([[b0, b1]]),
+Matrix([[0]]))
+>>> ssm2 = StateSpaceModel(
+...         Matrix([[0, 1, 0],
+...                 [0, 0, 1],
+...                 [-a0, -a1, -a2]]),
+...         Matrix([0, 0, 1]),
+...         Matrix([[b0, b1, b2]]))
+>>> ssm2
+StateSpaceModel(
+Matrix([
+[  0,   1,   0],
+[  0,   0,   1],
+[-a0, -a1, -a2]]),
+Matrix([
+[0],
+[0],
+[1]]),
+Matrix([[b0, b1, b2]]),
+Matrix([[0]]))
+>>> ssm_series = ssm1.cascade(ssm2)
+>>> ssm_series
+StateSpaceModel(
+Matrix([
+[  0,   1,   0,   0,   0],
+[-a0, -a1,   0,   0,   0],
+[  0,   0,   0,   1,   0],
+[  0,   0,   0,   0,   1],
+[ b0,  b1, -a0, -a1, -a2]]),
+Matrix([
+[0],
+[1],
+[0],
+[0],
+[0]]),
+Matrix([[0, 0, b0, b1, b2]]),
+Matrix([[0]]))
+>>> ssm_parallel = ssm2.parallel(ssm2)
+>>> ssm_parallel
+StateSpaceModel(
+Matrix([
+[  0,   1,   0,   0,   0,   0],
+[  0,   0,   1,   0,   0,   0],
+[-a0, -a1, -a2,   0,   0,   0],
+[  0,   0,   0,   0,   1,   0],
+[  0,   0,   0,   0,   0,   1],
+[  0,   0,   0, -a0, -a1, -a2]]),
+Matrix([
+[0],
+[0],
+[1],
+[0],
+[0],
+[1]]),
+Matrix([[b0, b1, b2, b0, b1, b2]]),
+Matrix([[0]]))
+
+>>> var('a, b, s')
+(a, b, s)
+>>> tfm1 = TransferFunctionModel(Matrix([1 / (a * s)]))
+>>> tfm1
+TransferFunctionModel(Matrix([[1/(a*s)]]))
+>>> tfm2 = TransferFunctionModel(Matrix([(b + s) / (a + s)]))
+>>> tfm2
+TransferFunctionModel(Matrix([[(b + s)/(a + s)]]))
+>>> tfm_series = tfm1.cascade(tfm2)
+>>> tfm_series
+TransferFunctionModel(Matrix([[(b + s)/(a*s*(a + s))]]))
+>>> tfm_parallel = tfm1.cascade(tfm2)
+>>> tfm_parallel
+TransferFunctionModel(Matrix([[(b + s)/(a*s*(a + s))]]))
+
+Note that the systems must have a matching nubmer of inputs and outputs for this to work!
+
+Evaluating models
+=================
+
+Models are always evaluated using the `evaluate` method.
+It is done numercialy or symbolicaly, depending on the parameters of the call:
+
+>>> from sympy import Heaviside
+>>> var('t') # symbolic evaluation
+>>> u = Matrix([Heaviside(t)])
+>>> x0 = zeros(3, 1)
+>>> t0 = 0
+>>> y = ssm.evaluate(u, x0, t, t0)
+>>> y
+Matrix([
+[     t],
+[sin(t)]])
+
+>>> y_list = ssm.evaluate(u, x0, (t, [0, 10, 0.1]))
+
+
+Note that the only difference between the calls is that in the second one, t is substituted by `(t, np.arange(0, 10, 0.1))`; a tuple of the symbol t and a list of times to evaluate for.
+The first call will return a matrix valued expression in terms of t.
+The second call will return a list of matrix valued outputs, matching the time input.
```

Comment:
SymPy uses two approaches to turn things from symbols to numerics: `.evalf()` and `lambdify()` we should follow the same patterns here. An `evaluate()` method is not following sympy standards.

Reply:
So should I rename `evaluate` as `evalf`?

Reply:
It doesn't have the same signature as evalf so it shouldn't be renamed to that. The method can just return a symbolic object and the documentation here can advise how to use either evalf or lambidfy to turn that into the other forms of output.

I think that the API here isn't quite right. Basically we have something that produces ODEs and this method gives numerical or analytic solutions to those ODEs. This should really be done using `dsolve` though rather than implementing specific code here to solve this kind of ODE. Especially I don't think that there should be code here for numerically solving these ODEs if that isn't something that we have in dsolve

Reply:
I'm not sure dsolve is the appropriate solution. As soon as the state matrix A is 6x6 there are no analytical solutions for the eigenvalues and the solutions for 4x4 and 5x5 are too complicated to be useful. Solving the ODEs shouldn't be the purpose of this module. The purpose is to manipulate and analyses the ODEs using the methods in control theory methods. Solving should almost always be done numerically.

Reply:
See #18023